### ISSUE TITLE
New release (3.0?)

### ISSUE THREAD

**Issue by itsdalmo** [buildkite/lifecycled#53]
I think we might be ready for a new release since merging the following PRs:
- https://github.com/buildkite/lifecycled/pull/45
- https://github.com/buildkite/lifecycled/pull/52

Perhaps a 3.0, since technically, the public interface has changed with the addition of a `--spot` flag? Thoughts @lox?

**Comment by lox** [comment]
Yeah, I was going to do a bit of testing and get #49 merged and then put out a 3.0!

**Comment by itsdalmo** [comment]
Have you had a chance to do any tests @lox?

I ask because I'm working on a deployment using lifecycled `v2.1.1` (Concourse), and dangling queues are troubling me. The current version does not clean up the queue if the handler script exits with an error, since [`complete()` is never reached in this case](https://github.com/buildkite/lifecycled/blob/v2.1.1/daemon.go#L167).

**Comment by lox** [comment]
Hmmmm. I haven't, apologies. I'll prioritize it if it's still an issue in the 2.x series.

**Comment by lox** [comment]
Ok, I did some testing today. I reckon we should have spot enabled by default and add a `--no-spot` flag to opt out. Thoughts @itsdalmo?

**Comment by itsdalmo** [comment]
Roger that! I don't use `lifecycled` with spot instances myself since the handler can take more >2min to execute, but for backwards compatibility it makes sense to inverse the logic 👍

**Comment by itsdalmo** [comment]
@lox - are you still testing the current master branch?

**Comment by lox** [comment]
Sorry, I finished testing, everything looks good. Let's put out a release!
